spec/api/keys/keys_spec.rb

```diff
@@ -818,51 +835,102 @@ def make_user_payload(payload)
         end
       end
     end
-    context "listing keys" do
+
+    context "listing key(s)" do
       context "when multiple keys are present" do
         context "for a client" do
           before(:each) do
             post("#{org_base_url}/clients", superuser, :payload => org_client_payload).should look_like({:status => 201})
-            add_client_key($org['name'], org_client_name, :key, "key1", :expires => "2017-12-24T21:00" ).should look_like({:status=>201})
-            add_client_key($org['name'], org_client_name, :alt_key, "key2", :expires => "2012-01-01T00:00" ).should look_like({:status=>201})
+            add_client_key($org['name'], org_client_name, :key, "key1", :expires =>  unexpired_date).should look_like({:status=>201})
+            add_client_key($org['name'], org_client_name, :alt_key, "key2", :expires => expired_date).should look_like({:status=>201})
           end
           after(:each) do
             delete("#{org_base_url}/clients/#{org_client_name}", superuser).should look_like({:status => 200})
           end
-          it "all keys should be listed with correct expiry indicators" do
-            list_client_keys($org['name'], org_client_name, superuser).should look_like({
-              :status => 200,
-              :body => [
-                { "name" => "default", "uri" => "#{org_base_url}/clients/#{org_client_name}/keys/default", "expired" => false },
-                { "name" => "key1", "uri" => "#{org_base_url}/clients/#{org_client_name}/keys/key1", "expired" => false },
-                { "name" => "key2", "uri" => "#{org_base_url}/clients/#{org_client_name}/keys/key2", "expired" => true}
-              ]})
-
+          context "when GET /organizations/org/clients/client/keys is called (list keys)" do
```

Comment:
Well part of that is the original context. Every context block string should start with "when". It makes things much more readable and that is not the case in the original contexts.